### DISCUSSION TITLE
Insert to item to database table on button click

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5688] [Help]
Hello,

I want to add a button in grid form besides column picker and wanted to add item in database table by checking if item do not exists than insert and refresh the grid.
Say,
I want to create and insert a user in database table on button click
where button is present in grid form and no dialog on button click for insert should be pop-up it should insert hard coded values automatically and also need to reflect in grid after insertion.

**Comment by edwardch** [comment]
Try this in your XyzGrid.ts

```ts
protected getButtons(): Serenity.ToolButton[] {

	var buttons = super.getButtons();

	if (Authorization.hasPermission("Administration:Security")) {
		buttons.push({
			title: 'Insert Record',
			cssClass: 'add-button',
			onClick: () => {
				var entity: XyzRow;
				// Set up your entity properties here
				// ...

				XyzService.Create(
					{
						Entity: entity
					},
					response => {
						this.refresh();
					}
				);
			}
		});    	
	}

	return buttons;
}
```

**Reply by TechAspirant** [reply]
How to check If data do exists, If no than only insert.

**Reply by TechAspirant** [reply]
Also, It results entity as undefined.

**Reply by edwardch** [reply]
> How to check If data do exists, If no than only insert.

Try something like this
```ts
// Change this to whatever ID you want
var xyzId = 123;

var entity = XyzModule.XyzService.Retrieve({
	EntityId: xyzId
}, response => {
        // You might want to check this syntax whether the entity exists or not
	if (!response.Entity) {
		// add your insert code here
	}
});
```

> Also, It results entity as undefined.

Yes I added this comment so you could assign values / create your own new entity
```ts
// Set up your entity properties here
// ...
```

**Reply by TechAspirant** [reply]
Can I use other column instead of EntityId

**Reply by edwardch** [reply]
Look in the issues to see where people have used `XyzService.List(`

Then in the response check if there are entities in the result

**Reply by TechAspirant** [reply]
Thanks, Can I use XyzService.Create in for loop in my case its not waiting for first entry to create
